lib/openstudio-standards/exterior_lighting/information.rb

```diff
@@ -0,0 +1,185 @@
+          building_type_hash[:number_of_beds] = hash[:number_of_beds]
+          building_type_hashes[hash[:standards_building_type]] = building_type_hash
+        end
+      end
+
+      # rename Office to SmallOffice, MediumOffice or LargeOffice depending on size
+      if building_type_hashes.key?('Office')
+        floor_area = building_type_hashes['Office'][:floor_area]
+        office_type = if floor_area < 2750
+                        'SmallOffice'
+                      elsif floor_area < 25_250
+                        'MediumOffice'
+                      else
+                        'LargeOffice'
+                      end
+        building_type_hashes[office_type] = building_type_hashes.delete('Office')
+      end
+
+      # initialize parking areas and drives area variables
+      parking_area_and_drives_area = 0.0
+      main_entries = 0.0
+      other_doors = 0.0
+      rollup_doors = 0.0
+      drive_through_windows = 0.0
+      canopy_entry_area = 0.0
+      canopy_emergency_area = 0.0
+      ground_story_ext_wall_area = 0.0
+
+      # temporary std for model_effective_num_stories method
+      std = Standard.build('90.1-2013')
+
+      # calculate exterior lighting properties for each building type
+      building_type_hashes.each do |building_type, hash|
+        # calculate floor area and ground floor area in IP units
+        floor_area_ft2 = OpenStudio.convert(hash[:floor_area], 'm^2', 'ft^2').get
+        effective_num_stories = std.model_effective_num_stories(model)
+        ground_floor_area_ft2 = floor_area_ft2 / effective_num_stories[:above_grade]
+
+        # load parking area properties for standards building type
+        parking_properties = parking_hsh.select { |h| h[:building_type] == building_type }
+
+        if parking_properties.nil? || parking_properties.empty?
+          OpenStudio.logFree(OpenStudio::Error, 'openstudio.prototype.ExteriorLighting', "Could not find parking data for #{building_type}.")
+          return {}
+        end
+        parking_properties = parking_properties[0]
+
+        # calculate number of parking spots
+        num_spots = 0.0
+        if !parking_properties[:building_area_per_spot].nil?
+          num_spots += floor_area_ft2 / parking_properties[:building_area_per_spot].to_f
+        elsif !parking_properties[:units_per_spot].nil?
+          num_spots += hash[:number_of_units] / parking_properties[:units_per_spot].to_f
+        elsif !parking_properties[:students_per_spot].nil?
+          num_spots += hash[:number_of_students] / parking_properties[:students_per_spot].to_f
+        elsif !parking_properties[:beds_per_spot].nil?
+          num_spots += hash[:number_of_beds] / parking_properties[:beds_per_spot].to_f
+        else
+          OpenStudio.logFree(OpenStudio::Info, 'openstudio.prototype.exterior_lights', "Unexpected key, can't calculate number of parking spots from #{parking_properties.keys.first}.")
+        end
```

Comment:
I don't think this log message makes sense here - it would only trigger if all the other `parking_properties` hash values are nil, not that an unexpected key was encountered. Plus, in that case, `parking_properties.keys.first` isn't necessarily the 'unexpected' key.

Reply:
removed statement